### PR TITLE
feat: xml support

### PR DESCRIPTION
**Pull request by simlu**

### CODE REVIEW

File: src/util/xml-merge.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
+const objectDeepContain = require('object-deep-contain');
+
+const mergeRec = (target, changeset) => Object
+  .entries(Object.assign(target.type === 'element' ? { elements: [] } : {}, target))
+  .map(([attr, elements]) => {
+    const changesetElements = changeset[attr];
+
+    if (
+      changesetElements === undefined
+      || !Array.isArray(elements)
```
Comment (MacMcIrish) [inline src/util/xml-merge.js]: One line?

File: src/util/xml-merge.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    ) {
+      return [attr, changesetElements !== undefined ? changesetElements : elements];
+    }
+
+    let next = 0;
+    for (let idx = 0, len = elements.length; idx < len && next < changesetElements.length; idx += 1) {
```
Comment (MacMcIrish) [inline src/util/xml-merge.js]: Seeing as the length of `elements` isn't modified below:
```suggestion
    for (let idx = 0; idx <  elements.length && next < changesetElements.length; idx += 1) {
```

Comment (simlu) [reply src/util/xml-merge.js]: Well, the argument is actually the other way around... 

but performance doesn't matter too much and it's shorter.

**Pull request merged**